Unable to link on Windows

Hi Tomas,

I'm working through your roguelike tutorial and just added the first window creation code from part 1. The project won't build now. I'm getting a linker 1120 error.  I'm also seeing the same linker error when I try to run an example from this repo using `cargo build --release ; cargo run --example keyboard`

I'm on Win7 and using the MSVC compiler. Cargo version is 0.16.0-nightly. Any ideas?

```
$ cargo build --release --verbose
   Compiling libc v0.2.21
   Compiling gcc v0.3.43
   Compiling bitflags v0.1.1
   Compiling lazy_static v0.1.16
   Compiling pkg-config v0.3.9
     Running `rustc --crate-name libc C:\\Users\\michael\\.cargo\\registry\\src\\github.com-1ecc6299db9ec823\\libc-0.2.21\\src\\lib.rs --crate-type lib -C opt-level=3 --cfg feature=\"default\" --cfg feature=\"use_std\" -C metadata=9ac948aa14b90e38 -C extra-filename=-9ac948aa14b90e38 --out-dir C:\\dev\\games\\rusty-roguelike\\target\\release\\deps --emit=dep-info,link -L dependency=C:\\dev\\games\\rusty-roguelike\\target\\release\\deps --cap-lints allow`
     Running `rustc --crate-name gcc C:\\Users\\michael\\.cargo\\registry\\src\\github.com-1ecc6299db9ec823\\gcc-0.3.43\\src\\lib.rs --crate-type lib -C opt-level=3 -C metadata=62787f88cb47e19d -C extra-filename=-62787f88cb47e19d --out-dir C:\\dev\\games\\rusty-roguelike\\target\\release\\deps --emit=dep-info,link -L dependency=C:\\dev\\games\\rusty-roguelike\\target\\release\\deps --cap-lints allow`
     Running `rustc --crate-name lazy_static C:\\Users\\michael\\.cargo\\registry\\src\\github.com-1ecc6299db9ec823\\lazy_static-0.1.16\\src\\lib.rs --crate-type lib -C opt-level=3 -C metadata=8dc5aa0427cd4b28 -C extra-filename=-8dc5aa0427cd4b28 --out-dir C:\\dev\\games\\rusty-roguelike\\target\\release\\deps --emit=dep-info,link -L dependency=C:\\dev\\games\\rusty-roguelike\\target\\release\\deps --cap-lints allow`
     Running `rustc --crate-name pkg_config C:\\Users\\michael\\.cargo\\registry\\src\\github.com-1ecc6299db9ec823\\pkg-config-0.3.9\\src\\lib.rs --crate-type lib -C opt-level=3 -C metadata=475156e224ad6f12 -C extra-filename=-475156e224ad6f12 --out-dir C:\\dev\\games\\rusty-roguelike\\target\\release\\deps --emit=dep-info,link -L dependency=C:\\dev\\games\\rusty-roguelike\\target\\release\\deps --cap-lints allow`
     Running `rustc --crate-name bitflags C:\\Users\\michael\\.cargo\\registry\\src\\github.com-1ecc6299db9ec823\\bitflags-0.1.1\\src\\lib.rs --crate-type lib -C opt-level=3 -C metadata=7f6500b20ee17639 -C extra-filename=-7f6500b20ee17639 --out-dir C:\\dev\\games\\rusty-roguelike\\target\\release\\deps --emit=dep-info,link -L dependency=C:\\dev\\games\\rusty-roguelike\\target\\release\\deps --cap-lints allow`
   Compiling rand v0.3.15
     Running `rustc --crate-name rand C:\\Users\\michael\\.cargo\\registry\\src\\github.com-1ecc6299db9ec823\\rand-0.3.15\\src\\lib.rs --crate-type lib -C opt-level=3 -C metadata=f3e5211f56db4055 -C extra-filename=-f3e5211f56db4055 --out-dir C:\\dev\\games\\rusty-roguelike\\target\\release\\deps --emit=dep-info,link -L dependency=C:\\dev\\games\\rusty-roguelike\\target\\release\\deps --extern libc=C:\\dev\\games\\rusty-roguelike\\target\\release\\deps\\liblibc-9ac948aa14b90e38.rlib --cap-lints allow`
   Compiling tcod-sys v4.0.1
     Running `rustc --crate-name build_script_build C:\\Users\\michael\\.cargo\\registry\\src\\github.com-1ecc6299db9ec823\\tcod-sys-4.0.1\\build.rs --crate-type bin -C opt-level=3 -C metadata=4400103e18cdc123 -C extra-filename=-4400103e18cdc123 --out-dir C:\\dev\\games\\rusty-roguelike\\target\\release\\build\\tcod-sys-4400103e18cdc123 --emit=dep-info,link -L dependency=C:\\dev\\games\\rusty-roguelike\\target\\release\\deps --extern pkg_config=C:\\dev\\games\\rusty-roguelike\\target\\release\\deps\\libpkg_config-475156e224ad6f12.rlib --extern gcc=C:\\dev\\games\\rusty-roguelike\\target\\release\\deps\\libgcc-62787f88cb47e19d.rlib --cap-lints allow`
     Running `C:\dev\games\rusty-roguelike\target\release\build\tcod-sys-4400103e18cdc123\build-script-build`
     Running `rustc --crate-name tcod_sys C:\\Users\\michael\\.cargo\\registry\\src\\github.com-1ecc6299db9ec823\\tcod-sys-4.0.1\\lib.rs --crate-type lib -C opt-level=3 -C metadata=7d36bd8b50cc5174 -C extra-filename=-7d36bd8b50cc5174 --out-dir C:\\dev\\games\\rusty-roguelike\\target\\release\\deps --emit=dep-info,link -L dependency=C:\\dev\\games\\rusty-roguelike\\target\\release\\deps --extern libc=C:\\dev\\games\\rusty-roguelike\\target\\release\\deps\\liblibc-9ac948aa14b90e38.rlib --cap-lints allow -L native=X:\\Programs\\Visual\ Studio\ 15\\VC\\atlmfc\\lib\\amd64 -L native=C:\\dev\\games\\rusty-roguelike\\target\\release\\build\\tcod-sys-ee576a2106520bdb\\out -L C:\\dev\\games\\rusty-roguelike\\target\\release\\build\\tcod-sys-ee576a2106520bdb\\out -L C:\\dev\\games\\rusty-roguelike\\target\\release\\build\\tcod-sys-ee576a2106520bdb\\out -l static=z -l dylib=SDL.dll -l User32 -l dylib=tcod`
   Compiling tcod v0.10.0
     Running `rustc --crate-name tcod C:\\Users\\michael\\.cargo\\registry\\src\\github.com-1ecc6299db9ec823\\tcod-0.10.0\\src/lib.rs --crate-type lib -C opt-level=3 -C metadata=c4ab14fb7f48b4a1 -C extra-filename=-c4ab14fb7f48b4a1 --out-dir C:\\dev\\games\\rusty-roguelike\\target\\release\\deps --emit=dep-info,link -L dependency=C:\\dev\\games\\rusty-roguelike\\target\\release\\deps --extern libc=C:\\dev\\games\\rusty-roguelike\\target\\release\\deps\\liblibc-9ac948aa14b90e38.rlib --extern tcod_sys=C:\\dev\\games\\rusty-roguelike\\target\\release\\deps\\libtcod_sys-7d36bd8b50cc5174.rlib --extern bitflags=C:\\dev\\games\\rusty-roguelike\\target\\release\\deps\\libbitflags-7f6500b20ee17639.rlib --extern lazy_static=C:\\dev\\games\\rusty-roguelike\\target\\release\\deps\\liblazy_static-8dc5aa0427cd4b28.rlib --cap-lints allow -L native=X:\\Programs\\Visual\ Studio\ 15\\VC\\atlmfc\\lib\\amd64 -L native=C:\\dev\\games\\rusty-roguelike\\target\\release\\build\\tcod-sys-ee576a2106520bdb\\out -L C:\\dev\\games\\rusty-roguelike\\target\\release\\build\\tcod-sys-ee576a2106520bdb\\out -L C:\\dev\\games\\rusty-roguelike\\target\\release\\build\\tcod-sys-ee576a2106520bdb\\out`
   Compiling rusty-roguelike v0.1.0 (file:///C:/dev/games/rusty-roguelike)
     Running `rustc --crate-name rusty_roguelike src\\main.rs --crate-type bin -C opt-level=3 -C metadata=bcf554dbd8bb6d21 -C extra-filename=-bcf554dbd8bb6d21 --out-dir C:\\dev\\games\\rusty-roguelike\\target\\release\\deps --emit=dep-info,link -L dependency=C:\\dev\\games\\rusty-roguelike\\target\\release\\deps --extern rand=C:\\dev\\games\\rusty-roguelike\\target\\release\\deps\\librand-f3e5211f56db4055.rlib --extern tcod=C:\\dev\\games\\rusty-roguelike\\target\\release\\deps\\libtcod-c4ab14fb7f48b4a1.rlib -L native=X:\\Programs\\Visual\ Studio\ 15\\VC\\atlmfc\\lib\\amd64 -L native=C:\\dev\\games\\rusty-roguelike\\target\\release\\build\\tcod-sys-ee576a2106520bdb\\out -L C:\\dev\\games\\rusty-roguelike\\target\\release\\build\\tcod-sys-ee576a2106520bdb\\out -L C:\\dev\\games\\rusty-roguelike\\target\\release\\build\\tcod-sys-ee576a2106520bdb\\out`

error: linking with `link.exe` failed: exit code: 1120
  |
  = note: "X:\\Programs\\Visual Studio 15\\VC\\bin\\amd64\\link.exe" "/LIBPATH:X:\\Programs\\Visual Studio 15\\VC\\lib\\amd64" "/LIBPATH:C:\\Program Files (x86)\\Windows Kits\\10\\lib\\10.0.10240.0\\ucrt\\x64" "/LIBPATH:C:\\Program Files (x86)\\Windows Kits\\8.1\\lib\\winv6.3\\um\\x64" "/NOLOGO" "/NXCOMPAT" "/LIBPATH:C:\\Users\\michael\\.rustup\\toolchains\\stable-x86_64-pc-windows-msvc\\lib\\rustlib\\x86_64-pc-windows-msvc\\lib" "C:\\dev\\games\\rusty-roguelike\\target\\release\\deps\\rusty_roguelike-bcf554dbd8bb6d21.0.o" "/OUT:C:\\dev\\games\\rusty-roguelike\\target\\release\\deps\\rusty_roguelike-bcf554dbd8bb6d21.exe" "/OPT:REF,ICF" "/DEBUG" "/LIBPATH:C:\\dev\\games\\rusty-roguelike\\target\\release\\deps" "/LIBPATH:X:\\Programs\\Visual Studio 15\\VC\\atlmfc\\lib\\amd64" "/LIBPATH:C:\\dev\\games\\rusty-roguelike\\target\\release\\build\\tcod-sys-ee576a2106520bdb\\out" "/LIBPATH:C:\\dev\\games\\rusty-roguelike\\target\\release\\build\\tcod-sys-ee576a2106520bdb\\out" "/LIBPATH:C:\\dev\\games\\rusty-roguelike\\target\\release\\build\\tcod-sys-ee576a2106520bdb\\out" "/LIBPATH:C:\\Users\\michael\\.rustup\\toolchains\\stable-x86_64-pc-windows-msvc\\lib\\rustlib\\x86_64-pc-windows-msvc\\lib" "C:\\dev\\games\\rusty-roguelike\\target\\release\\deps\\libtcod-c4ab14fb7f48b4a1.rlib" "C:\\dev\\games\\rusty-roguelike\\target\\release\\deps\\libbitflags-7f6500b20ee17639.rlib" "C:\\dev\\games\\rusty-roguelike\\target\\release\\deps\\libtcod_sys-7d36bd8b50cc5174.rlib" "C:\\dev\\games\\rusty-roguelike\\target\\release\\deps\\liblibc-9ac948aa14b90e38.rlib" "C:\\dev\\games\\rusty-roguelike\\target\\release\\deps\\liblazy_static-8dc5aa0427cd4b28.rlib" "C:\\Users\\michael\\.rustup\\toolchains\\stable-x86_64-pc-windows-msvc\\lib\\rustlib\\x86_64-pc-windows-msvc\\lib\\libstd-259356a62c16e209.rlib" "C:\\Users\\michael\\.rustup\\toolchains\\stable-x86_64-pc-windows-msvc\\lib\\rustlib\\x86_64-pc-windows-msvc\\lib\\libpanic_unwind-6090925440afc6a0.rlib" "C:\\Users\\michael\\.rustup\\toolchains\\stable-x86_64-pc-windows-msvc\\lib\\rustlib\\x86_64-pc-windows-msvc\\lib\\libunwind-4de2657775b53c86.rlib" "C:\\Users\\michael\\.rustup\\toolchains\\stable-x86_64-pc-windows-msvc\\lib\\rustlib\\x86_64-pc-windows-msvc\\lib\\liblibc-726a7bfd60260b09.rlib" "C:\\Users\\michael\\.rustup\\toolchains\\stable-x86_64-pc-windows-msvc\\lib\\rustlib\\x86_64-pc-windows-msvc\\lib\\librand-626ae980c5da662e.rlib" "C:\\Users\\michael\\.rustup\\toolchains\\stable-x86_64-pc-windows-msvc\\lib\\rustlib\\x86_64-pc-windows-msvc\\lib\\libcollections-9c298ff08fd36373.rlib" "C:\\Users\\michael\\.rustup\\toolchains\\stable-x86_64-pc-windows-msvc\\lib\\rustlib\\x86_64-pc-windows-msvc\\lib\\liballoc-2c53774924f65757.rlib" "C:\\Users\\michael\\.rustup\\toolchains\\stable-x86_64-pc-windows-msvc\\lib\\rustlib\\x86_64-pc-windows-msvc\\lib\\liballoc_system-dceaab434bd30f7c.rlib" "C:\\Users\\michael\\.rustup\\toolchains\\stable-x86_64-pc-windows-msvc\\lib\\rustlib\\x86_64-pc-windows-msvc\\lib\\libstd_unicode-395a85a6bc8d70ac.rlib" "C:\\Users\\michael\\.rustup\\toolchains\\stable-x86_64-pc-windows-msvc\\lib\\rustlib\\x86_64-pc-windows-msvc\\lib\\libcore-e714f4bfba22deac.rlib" "C:\\Users\\michael\\.rustup\\toolchains\\stable-x86_64-pc-windows-msvc\\lib\\rustlib\\x86_64-pc-windows-msvc\\lib\\libcompiler_builtins-0c9c28ea236973f1.rlib" "tcod.lib" "tcod.lib" "SDL.dll.lib" "User32.lib" "advapi32.lib" "ws2_32.lib" "userenv.lib" "shell32.lib" "msvcrt.lib"
  = note: libtcod-c4ab14fb7f48b4a1.rlib(tcod-c4ab14fb7f48b4a1.0.o) : error LNK2019: unresolved external symbol __imp_TCOD_bsp_new_with_size referenced in function _ZN4tcod3bsp3Bsp13new_with_size17had6c75491875e31dE
libtcod-c4ab14fb7f48b4a1.rlib(tcod-c4ab14fb7f48b4a1.0.o) : error LNK2019: unresolved external symbol __imp_TCOD_bsp_remove_sons referenced in function _ZN4tcod3bsp3Bsp11remove_sons17h122fd0484b7cf322E
libtcod-c4ab14fb7f48b4a1.rlib(tcod-c4ab14fb7f48b4a1.0.o) : error LNK2019: unresolved external symbol __imp_TCOD_bsp_split_once referenced in function _ZN4tcod3bsp3Bsp10split_once17h2099b52f80df069cE
libtcod-c4ab14fb7f48b4a1.rlib(tcod-c4ab14fb7f48b4a1.0.o) : error LNK2019: unresolved external symbol __imp_TCOD_random_get_instance referenced in function _ZN4tcod3bsp3Bsp15split_recursive17h789c8e1297e4e9c9E
libtcod-c4ab14fb7f48b4a1.rlib(tcod-c4ab14fb7f48b4a1.0.o) : error LNK2019: unresolved external symbol __imp_TCOD_bsp_split_recursive referenced in function _ZN4tcod3bsp3Bsp15split_recursive17h789c8e1297e4e9c9E
libtcod-c4ab14fb7f48b4a1.rlib(tcod-c4ab14fb7f48b4a1.0.o) : error LNK2019: unresolved external symbol __imp_TCOD_random_delete referenced in function _ZN4tcod3bsp3Bsp15split_recursive17h789c8e1297e4e9c9E
libtcod-c4ab14fb7f48b4a1.rlib(tcod-c4ab14fb7f48b4a1.0.o) : error LNK2019: unresolved external symbol __imp_TCOD_bsp_resize referenced in function _ZN4tcod3bsp3Bsp6resize17heee6d62cf83b78f2E
libtcod-c4ab14fb7f48b4a1.rlib(tcod-c4ab14fb7f48b4a1.0.o) : error LNK2019: unresolved external symbol __imp_TCOD_bsp_left referenced in function _ZN4tcod3bsp3Bsp4left17h8a242240799621d6E
libtcod-c4ab14fb7f48b4a1.rlib(tcod-c4ab14fb7f48b4a1.0.o) : error LNK2019: unresolved external symbol __imp_TCOD_bsp_right referenced in function _ZN4tcod3bsp3Bsp5right17h3440815266b2eb24E
libtcod-c4ab14fb7f48b4a1.rlib(tcod-c4ab14fb7f48b4a1.0.o) : error LNK2019: unresolved external symbol __imp_TCOD_bsp_father referenced in function _ZN4tcod3bsp3Bsp6father17h1cc4a9ea656dce06E
libtcod-c4ab14fb7f48b4a1.rlib(tcod-c4ab14fb7f48b4a1.0.o) : error LNK2019: unresolved external symbol __imp_TCOD_bsp_is_leaf referenced in function _ZN4tcod3bsp3Bsp7is_leaf17h7a26486a1692e529E
libtcod-c4ab14fb7f48b4a1.rlib(tcod-c4ab14fb7f48b4a1.0.o) : error LNK2019: unresolved external symbol __imp_TCOD_bsp_contains referenced in function _ZN4tcod3bsp3Bsp8contains17h2fc07e4ea049ce59E
libtcod-c4ab14fb7f48b4a1.rlib(tcod-c4ab14fb7f48b4a1.0.o) : error LNK2019: unresolved external symbol __imp_TCOD_bsp_find_node referenced in function _ZN4tcod3bsp3Bsp9find_node17h904813180eca2679E
libtcod-c4ab14fb7f48b4a1.rlib(tcod-c4ab14fb7f48b4a1.0.o) : error LNK2019: unresolved external symbol __imp_TCOD_bsp_delete referenced in function _ZN64_$LT$tcod..bsp..Bsp$LT$$u27$a$GT$$u20$as$u20$core..ops..Drop$GT$4drop17he6aae349b1c9730cE
libtcod-c4ab14fb7f48b4a1.rlib(tcod-c4ab14fb7f48b4a1.0.o) : error LNK2019: unresolved external symbol __imp_TCOD_color_HSV referenced in function _ZN4tcod6colors5Color12new_from_hsv17h1c270f423baa09c8E
libtcod-c4ab14fb7f48b4a1.rlib(tcod-c4ab14fb7f48b4a1.0.o) : error LNK2019: unresolved external symbol __imp_TCOD_color_get_HSV referenced in function _ZN4tcod6colors5Color3hsv17hf8a65a02bc345562E
libtcod-c4ab14fb7f48b4a1.rlib(tcod-c4ab14fb7f48b4a1.0.o) : error LNK2019: unresolved external symbol __imp_TCOD_color_shift_hue referenced in function _ZN4tcod6colors5Color9shift_hue17h0bbfe16192e58bbaE
libtcod-c4ab14fb7f48b4a1.rlib(tcod-c4ab14fb7f48b4a1.0.o) : error LNK2019: unresolved external symbol __imp_TCOD_color_scale_HSV referenced in function _ZN4tcod6colors5Color9scale_hsv17h92c41a95320b9aa0E
libtcod-c4ab14fb7f48b4a1.rlib(tcod-c4ab14fb7f48b4a1.0.o) : error LNK2019: unresolved external symbol __imp_TCOD_color_lerp referenced in function _ZN4tcod6colors4lerp17h8851f4e649e00245E
libtcod-c4ab14fb7f48b4a1.rlib(tcod-c4ab14fb7f48b4a1.0.o) : error LNK2019: unresolved external symbol __imp_TCOD_console_delete referenced in function _ZN60_$LT$tcod..console..Offscreen$u20$as$u20$core..ops..Drop$GT$4drop17h32e7e5714e77d732E
libtcod-c4ab14fb7f48b4a1.rlib(tcod-c4ab14fb7f48b4a1.0.o) : error LNK2019: unresolved external symbol __imp_TCOD_console_new referenced in function _ZN4tcod7console9Offscreen3new17hd3a1991b71fdb35cE
libtcod-c4ab14fb7f48b4a1.rlib(tcod-c4ab14fb7f48b4a1.0.o) : error LNK2019: unresolved external symbol __imp_TCOD_console_is_fullscreen referenced in function _ZN4tcod7console4Root13is_fullscreen17hdc6b16f6f2a078efE
libtcod-c4ab14fb7f48b4a1.rlib(tcod-c4ab14fb7f48b4a1.0.o) : error LNK2019: unresolved external symbol __imp_TCOD_console_set_fullscreen referenced in function _ZN4tcod7console4Root14set_fullscreen17h47abcc38a09f9546E
libtcod-c4ab14fb7f48b4a1.rlib(tcod-c4ab14fb7f48b4a1.0.o) : error LNK2019: unresolved external symbol __imp_TCOD_console_set_keyboard_repeat referenced in function _ZN4tcod7console4Root19set_keyboard_repeat17hb6d772bb48a8df85E
libtcod-c4ab14fb7f48b4a1.rlib(tcod-c4ab14fb7f48b4a1.0.o) : error LNK2019: unresolved external symbol __imp_TCOD_console_disable_keyboard_repeat referenced in function _ZN4tcod7console4Root23disable_keyboard_repeat17h48b4d9d021a30470E
libtcod-c4ab14fb7f48b4a1.rlib(tcod-c4ab14fb7f48b4a1.0.o) : error LNK2019: unresolved external symbol __imp_TCOD_console_is_active referenced in function _ZN4tcod7console4Root9is_active17hf7027a49dd899757E
libtcod-c4ab14fb7f48b4a1.rlib(tcod-c4ab14fb7f48b4a1.0.o) : error LNK2019: unresolved external symbol __imp_TCOD_console_has_mouse_focus referenced in function _ZN4tcod7console4Root9has_focus17hbacaf11b0dc0de11E
libtcod-c4ab14fb7f48b4a1.rlib(tcod-c4ab14fb7f48b4a1.0.o) : error LNK2019: unresolved external symbol __imp_TCOD_console_get_fade referenced in function _ZN4tcod7console4Root8get_fade17hf221d5fb44e14631E
libtcod-c4ab14fb7f48b4a1.rlib(tcod-c4ab14fb7f48b4a1.0.o) : error LNK2019: unresolved external symbol __imp_TCOD_console_get_fading_color referenced in function _ZN4tcod7console4Root16get_fading_color17h2d1383ea7d060b86E
libtcod-c4ab14fb7f48b4a1.rlib(tcod-c4ab14fb7f48b4a1.0.o) : error LNK2019: unresolved external symbol __imp_TCOD_console_set_fade referenced in function _ZN4tcod7console4Root8set_fade17ha1e3706fc4813acdE
libtcod-c4ab14fb7f48b4a1.rlib(tcod-c4ab14fb7f48b4a1.0.o) : error LNK2019: unresolved external symbol __imp_TCOD_console_wait_for_keypress referenced in function _ZN4tcod7console4Root17wait_for_keypress17h4f50079b0a6f8ae4E
libtcod-c4ab14fb7f48b4a1.rlib(tcod-c4ab14fb7f48b4a1.0.o) : error LNK2019: unresolved external symbol __imp_TCOD_console_check_for_keypress referenced in function _ZN4tcod7console4Root18check_for_keypress17h7b27cc5fe2e25e16E
libtcod-c4ab14fb7f48b4a1.rlib(tcod-c4ab14fb7f48b4a1.0.o) : error LNK2019: unresolved external symbol __imp_TCOD_console_is_window_closed referenced in function _ZN4tcod7console4Root13window_closed17h48b69b18d75dee57E
libtcod-c4ab14fb7f48b4a1.rlib(tcod-c4ab14fb7f48b4a1.0.o) : error LNK2019: unresolved external symbol __imp_TCOD_console_flush referenced in function _ZN4tcod7console4Root5flush17h5bdb4d5876d876ccE
libtcod-c4ab14fb7f48b4a1.rlib(tcod-c4ab14fb7f48b4a1.0.o) : error LNK2019: unresolved external symbol __imp_TCOD_console_credits_render referenced in function _ZN4tcod7console4Root14render_credits17h6051a9f795a0af6aE
libtcod-c4ab14fb7f48b4a1.rlib(tcod-c4ab14fb7f48b4a1.0.o) : error LNK2019: unresolved external symbol __imp_TCOD_console_set_custom_font referenced in function _ZN4tcod7console15RootInitializer4init17h761f1b557330e564E
libtcod-c4ab14fb7f48b4a1.rlib(tcod-c4ab14fb7f48b4a1.0.o) : error LNK2019: unresolved external symbol __imp_TCOD_console_init_root referenced in function _ZN4tcod7console15RootInitializer4init17h761f1b557330e564E
libtcod-c4ab14fb7f48b4a1.rlib(tcod-c4ab14fb7f48b4a1.0.o) : error LNK2019: unresolved external symbol __imp_TCOD_image_get_size referenced in function _ZN97_$LT$tcod..image..Image$u20$as$u20$tcod..bindings..FromNative$LT$$BP$mut$u20$libc..c_void$GT$$GT$11from_native17h2561a7e51e53bdb2E
libtcod-c4ab14fb7f48b4a1.rlib(tcod-c4ab14fb7f48b4a1.0.o) : error LNK2019: unresolved external symbol __imp_TCOD_image_new referenced in function _ZN4tcod5image5Image3new17h937bd08ad61abd77E
libtcod-c4ab14fb7f48b4a1.rlib(tcod-c4ab14fb7f48b4a1.0.o) : error LNK2019: unresolved external symbol __imp_TCOD_image_get_pixel referenced in function _ZN4tcod5image5Image9get_pixel17hf9a117122ac1799bE
libtcod-c4ab14fb7f48b4a1.rlib(tcod-c4ab14fb7f48b4a1.0.o) : error LNK2019: unresolved external symbol __imp_TCOD_image_get_alpha referenced in function _ZN4tcod5image5Image9get_alpha17h3187950f239b8723E
libtcod-c4ab14fb7f48b4a1.rlib(tcod-c4ab14fb7f48b4a1.0.o) : error LNK2019: unresolved external symbol __imp_TCOD_image_is_pixel_transparent referenced in function _ZN4tcod5image5Image20is_pixel_transparent17h50ed44c4aa79f916E
libtcod-c4ab14fb7f48b4a1.rlib(tcod-c4ab14fb7f48b4a1.0.o) : error LNK2019: unresolved external symbol __imp_TCOD_image_get_mipmap_pixel referenced in function _ZN4tcod5image5Image16get_mipmap_pixel17hb5326bab0340c667E
libtcod-c4ab14fb7f48b4a1.rlib(tcod-c4ab14fb7f48b4a1.0.o) : error LNK2019: unresolved external symbol __imp_TCOD_image_set_key_color referenced in function _ZN4tcod5image5Image13set_key_color17h11b68e52821548e2E
libtcod-c4ab14fb7f48b4a1.rlib(tcod-c4ab14fb7f48b4a1.0.o) : error LNK2019: unresolved external symbol __imp_TCOD_image_clear referenced in function _ZN4tcod5image5Image5clear17h2f3f2d1212a4e22aE
libtcod-c4ab14fb7f48b4a1.rlib(tcod-c4ab14fb7f48b4a1.0.o) : error LNK2019: unresolved external symbol __imp_TCOD_image_put_pixel referenced in function _ZN4tcod5image5Image9put_pixel17h8820d619f23ae341E
libtcod-c4ab14fb7f48b4a1.rlib(tcod-c4ab14fb7f48b4a1.0.o) : error LNK2019: unresolved external symbol __imp_TCOD_image_scale referenced in function _ZN4tcod5image5Image5scale17h0501f1fac0e960edE
libtcod-c4ab14fb7f48b4a1.rlib(tcod-c4ab14fb7f48b4a1.0.o) : error LNK2019: unresolved external symbol __imp_TCOD_image_hflip referenced in function _ZN4tcod5image5Image5hflip17h6a8d147991a7f425E
libtcod-c4ab14fb7f48b4a1.rlib(tcod-c4ab14fb7f48b4a1.0.o) : error LNK2019: unresolved external symbol __imp_TCOD_image_vflip referenced in function _ZN4tcod5image5Image5vflip17hf8ad450a44df4a1dE
libtcod-c4ab14fb7f48b4a1.rlib(tcod-c4ab14fb7f48b4a1.0.o) : error LNK2019: unresolved external symbol __imp_TCOD_image_rotate90 referenced in function _ZN4tcod5image5Image8rotate9017hc1e81f9488980f44E
libtcod-c4ab14fb7f48b4a1.rlib(tcod-c4ab14fb7f48b4a1.0.o) : error LNK2019: unresolved external symbol __imp_TCOD_image_invert referenced in function _ZN4tcod5image5Image6invert17h0fedf76481154cb3E
libtcod-c4ab14fb7f48b4a1.rlib(tcod-c4ab14fb7f48b4a1.0.o) : error LNK2019: unresolved external symbol __imp_TCOD_mouse_show_cursor referenced in function _ZN4tcod5input11show_cursor17h8aebd0da8a5071c5E
libtcod-c4ab14fb7f48b4a1.rlib(tcod-c4ab14fb7f48b4a1.0.o) : error LNK2019: unresolved external symbol __imp_TCOD_mouse_is_cursor_visible referenced in function _ZN4tcod5input17is_cursor_visible17hbb91f4b6e87b8112E
libtcod-c4ab14fb7f48b4a1.rlib(tcod-c4ab14fb7f48b4a1.0.o) : error LNK2019: unresolved external symbol __imp_TCOD_mouse_move referenced in function _ZN4tcod5input11move_cursor17h52a11f704b3c9260E
libtcod-c4ab14fb7f48b4a1.rlib(tcod-c4ab14fb7f48b4a1.0.o) : error LNK2019: unresolved external symbol __imp_TCOD_sys_check_for_event referenced in function _ZN4tcod5input15check_for_event17h6d0b45080ee76485E
libtcod-c4ab14fb7f48b4a1.rlib(tcod-c4ab14fb7f48b4a1.0.o) : error LNK2019: unresolved external symbol __imp_TCOD_line_init_mt referenced in function _ZN4tcod4line4Line3new17h020a50d25989c506E
libtcod-c4ab14fb7f48b4a1.rlib(tcod-c4ab14fb7f48b4a1.0.o) : error LNK2019: unresolved external symbol __imp_TCOD_line_step_mt referenced in function _ZN4tcod4line4Line4step17hf35199e24d91a5b3E
libtcod-c4ab14fb7f48b4a1.rlib(tcod-c4ab14fb7f48b4a1.0.o) : error LNK2019: unresolved external symbol __imp_TCOD_map_new referenced in function _ZN4tcod3map3Map3new17ha204c770d2456084E
libtcod-c4ab14fb7f48b4a1.rlib(tcod-c4ab14fb7f48b4a1.0.o) : error LNK2019: unresolved external symbol __imp_TCOD_map_get_width referenced in function _ZN4tcod3map3Map4size17h1ce7ac29218dfc8dE
libtcod-c4ab14fb7f48b4a1.rlib(tcod-c4ab14fb7f48b4a1.0.o) : error LNK2019: unresolved external symbol __imp_TCOD_map_get_height referenced in function _ZN4tcod3map3Map4size17h1ce7ac29218dfc8dE
libtcod-c4ab14fb7f48b4a1.rlib(tcod-c4ab14fb7f48b4a1.0.o) : error LNK2019: unresolved external symbol __imp_TCOD_map_set_properties referenced in function _ZN4tcod3map3Map3set17heb125a65a195d145E
libtcod-c4ab14fb7f48b4a1.rlib(tcod-c4ab14fb7f48b4a1.0.o) : error LNK2019: unresolved external symbol __imp_TCOD_map_compute_fov referenced in function _ZN4tcod3map3Map11compute_fov17h15c1ec9f9100bf01E
libtcod-c4ab14fb7f48b4a1.rlib(tcod-c4ab14fb7f48b4a1.0.o) : error LNK2019: unresolved external symbol __imp_TCOD_map_is_in_fov referenced in function _ZN4tcod3map3Map9is_in_fov17h3b2260113a84506cE
libtcod-c4ab14fb7f48b4a1.rlib(tcod-c4ab14fb7f48b4a1.0.o) : error LNK2019: unresolved external symbol __imp_TCOD_map_is_walkable referenced in function _ZN4tcod3map3Map11is_walkable17hd997a8b285c97988E
libtcod-c4ab14fb7f48b4a1.rlib(tcod-c4ab14fb7f48b4a1.0.o) : error LNK2019: unresolved external symbol __imp_TCOD_map_delete referenced in function _ZN50_$LT$tcod..map..Map$u20$as$u20$core..ops..Drop$GT$4drop17h66b0d191a1ac534cE
libtcod-c4ab14fb7f48b4a1.rlib(tcod-c4ab14fb7f48b4a1.0.o) : error LNK2019: unresolved external symbol __imp_TCOD_namegen_destroy referenced in function _ZN58_$LT$tcod..namegen..Namegen$u20$as$u20$core..ops..Drop$GT$4drop17h83c3c38883c52bd6E
libtcod-c4ab14fb7f48b4a1.rlib(tcod-c4ab14fb7f48b4a1.0.o) : error LNK2019: unresolved external symbol __imp_TCOD_namegen_get_sets referenced in function _ZN4tcod7namegen7Namegen8get_sets17h89318d241f3f4797E
libtcod-c4ab14fb7f48b4a1.rlib(tcod-c4ab14fb7f48b4a1.0.o) : error LNK2019: unresolved external symbol __imp_TCOD_list_size referenced in function _ZN4tcod7namegen7Namegen8get_sets17h89318d241f3f4797E
libtcod-c4ab14fb7f48b4a1.rlib(tcod-c4ab14fb7f48b4a1.0.o) : error LNK2019: unresolved external symbol __imp_TCOD_list_get referenced in function _ZN4tcod7namegen7Namegen8get_sets17h89318d241f3f4797E
libtcod-c4ab14fb7f48b4a1.rlib(tcod-c4ab14fb7f48b4a1.0.o) : error LNK2019: unresolved external symbol __imp_TCOD_noise_set_type referenced in function _ZN4tcod5noise5Noise8set_type17hc0dfd669cd87b34bE
libtcod-c4ab14fb7f48b4a1.rlib(tcod-c4ab14fb7f48b4a1.0.o) : error LNK2019: unresolved external symbol __imp_TCOD_noise_delete referenced in function _ZN54_$LT$tcod..noise..Noise$u20$as$u20$core..ops..Drop$GT$4drop17h504db98c550f70c9E
libtcod-c4ab14fb7f48b4a1.rlib(tcod-c4ab14fb7f48b4a1.0.o) : error LNK2019: unresolved external symbol __imp_TCOD_noise_new referenced in function _ZN4tcod5noise16NoiseInitializer4init17hd904398b9c627df7E
libtcod-c4ab14fb7f48b4a1.rlib(tcod-c4ab14fb7f48b4a1.0.o) : error LNK2019: unresolved external symbol __imp_TCOD_path_delete referenced in function _ZN74_$LT$tcod..pathfinding..AStar$LT$$u27$a$GT$$u20$as$u20$core..ops..Drop$GT$4drop17ha4aeee51b2f8ad67E
libtcod-c4ab14fb7f48b4a1.rlib(tcod-c4ab14fb7f48b4a1.0.o) : error LNK2019: unresolved external symbol __imp_TCOD_path_new_using_map referenced in function _ZN4tcod11pathfinding5AStar12new_from_map17h3d43117fb376b24eE
libtcod-c4ab14fb7f48b4a1.rlib(tcod-c4ab14fb7f48b4a1.0.o) : error LNK2019: unresolved external symbol __imp_TCOD_path_compute referenced in function _ZN4tcod11pathfinding5AStar4find17h8cce787470a37cddE
libtcod-c4ab14fb7f48b4a1.rlib(tcod-c4ab14fb7f48b4a1.0.o) : error LNK2019: unresolved external symbol __imp_TCOD_path_walk referenced in function _ZN4tcod11pathfinding5AStar13walk_one_step17hb342d304ba826892E
libtcod-c4ab14fb7f48b4a1.rlib(tcod-c4ab14fb7f48b4a1.0.o) : error LNK2019: unresolved external symbol __imp_TCOD_path_reverse referenced in function _ZN4tcod11pathfinding5AStar7reverse17h15f42e895df2ce40E
libtcod-c4ab14fb7f48b4a1.rlib(tcod-c4ab14fb7f48b4a1.0.o) : error LNK2019: unresolved external symbol __imp_TCOD_path_get_origin referenced in function _ZN4tcod11pathfinding5AStar6origin17h0ac6222d83a284c6E
libtcod-c4ab14fb7f48b4a1.rlib(tcod-c4ab14fb7f48b4a1.0.o) : error LNK2019: unresolved external symbol __imp_TCOD_path_get_destination referenced in function _ZN4tcod11pathfinding5AStar11destination17h4043bfeb7ac3b6a4E
libtcod-c4ab14fb7f48b4a1.rlib(tcod-c4ab14fb7f48b4a1.0.o) : error LNK2019: unresolved external symbol __imp_TCOD_path_size referenced in function _ZN4tcod11pathfinding5AStar3get17h1c2c8a86d14bea04E
libtcod-c4ab14fb7f48b4a1.rlib(tcod-c4ab14fb7f48b4a1.0.o) : error LNK2019: unresolved external symbol __imp_TCOD_path_get referenced in function _ZN4tcod11pathfinding5AStar3get17h1c2c8a86d14bea04E
libtcod-c4ab14fb7f48b4a1.rlib(tcod-c4ab14fb7f48b4a1.0.o) : error LNK2019: unresolved external symbol __imp_TCOD_path_is_empty referenced in function _ZN4tcod11pathfinding5AStar8is_empty17h110bd7a13ba69f16E
libtcod-c4ab14fb7f48b4a1.rlib(tcod-c4ab14fb7f48b4a1.0.o) : error LNK2019: unresolved external symbol __imp_TCOD_dijkstra_delete referenced in function _ZN77_$LT$tcod..pathfinding..Dijkstra$LT$$u27$a$GT$$u20$as$u20$core..ops..Drop$GT$4drop17hfd3f81c26149f9dcE
libtcod-c4ab14fb7f48b4a1.rlib(tcod-c4ab14fb7f48b4a1.0.o) : error LNK2019: unresolved external symbol __imp_TCOD_dijkstra_new referenced in function _ZN4tcod11pathfinding8Dijkstra12new_from_map17hcf9cad00fc3cdaadE
libtcod-c4ab14fb7f48b4a1.rlib(tcod-c4ab14fb7f48b4a1.0.o) : error LNK2019: unresolved external symbol __imp_TCOD_dijkstra_compute referenced in function _ZN4tcod11pathfinding8Dijkstra12compute_grid17h8a3969fda8e6ab76E
libtcod-c4ab14fb7f48b4a1.rlib(tcod-c4ab14fb7f48b4a1.0.o) : error LNK2019: unresolved external symbol __imp_TCOD_dijkstra_path_set referenced in function _ZN4tcod11pathfinding8Dijkstra4find17h1ec71d9cfda58e6cE
libtcod-c4ab14fb7f48b4a1.rlib(tcod-c4ab14fb7f48b4a1.0.o) : error LNK2019: unresolved external symbol __imp_TCOD_dijkstra_path_walk referenced in function _ZN4tcod11pathfinding8Dijkstra13walk_one_step17h2fc6b277951cb835E
libtcod-c4ab14fb7f48b4a1.rlib(tcod-c4ab14fb7f48b4a1.0.o) : error LNK2019: unresolved external symbol __imp_TCOD_dijkstra_get_distance referenced in function _ZN4tcod11pathfinding8Dijkstra18distance_from_root17h0040104c11c05900E
libtcod-c4ab14fb7f48b4a1.rlib(tcod-c4ab14fb7f48b4a1.0.o) : error LNK2019: unresolved external symbol __imp_TCOD_dijkstra_reverse referenced in function _ZN4tcod11pathfinding8Dijkstra7reverse17h69661cb58936962bE
libtcod-c4ab14fb7f48b4a1.rlib(tcod-c4ab14fb7f48b4a1.0.o) : error LNK2019: unresolved external symbol __imp_TCOD_dijkstra_size referenced in function _ZN4tcod11pathfinding8Dijkstra3get17h633053a9f194c146E
libtcod-c4ab14fb7f48b4a1.rlib(tcod-c4ab14fb7f48b4a1.0.o) : error LNK2019: unresolved external symbol __imp_TCOD_dijkstra_get referenced in function _ZN4tcod11pathfinding8Dijkstra3get17h633053a9f194c146E
libtcod-c4ab14fb7f48b4a1.rlib(tcod-c4ab14fb7f48b4a1.0.o) : error LNK2019: unresolved external symbol __imp_TCOD_dijkstra_is_empty referenced in function _ZN4tcod11pathfinding8Dijkstra8is_empty17h5e8572753140caa9E
libtcod-c4ab14fb7f48b4a1.rlib(tcod-c4ab14fb7f48b4a1.0.o) : error LNK2019: unresolved external symbol __imp_TCOD_random_new referenced in function _ZN4tcod6random3Rng3new17h2695b21c7c828bd8E
libtcod-c4ab14fb7f48b4a1.rlib(tcod-c4ab14fb7f48b4a1.0.o) : error LNK2019: unresolved external symbol __imp_TCOD_random_new_from_seed referenced in function _ZN4tcod6random3Rng13new_with_seed17h5deca2ddd7d808feE
libtcod-c4ab14fb7f48b4a1.rlib(tcod-c4ab14fb7f48b4a1.0.o) : error LNK2019: unresolved external symbol __imp_TCOD_random_save referenced in function _ZN4tcod6random3Rng4save17h5f320e464efd5905E
libtcod-c4ab14fb7f48b4a1.rlib(tcod-c4ab14fb7f48b4a1.0.o) : error LNK2019: unresolved external symbol __imp_TCOD_random_restore referenced in function _ZN4tcod6random3Rng7restore17h331b74aa307fa90fE
libtcod-c4ab14fb7f48b4a1.rlib(tcod-c4ab14fb7f48b4a1.0.o) : error LNK2019: unresolved external symbol __imp_TCOD_random_set_distribution referenced in function _ZN4tcod6random3Rng16set_distribution17hcd38f1b022878301E
libtcod-c4ab14fb7f48b4a1.rlib(tcod-c4ab14fb7f48b4a1.0.o) : error LNK2019: unresolved external symbol __imp_TCOD_random_get_int referenced in function _ZN4tcod6random3Rng7get_int17h211c5cf21c975ce8E
libtcod-c4ab14fb7f48b4a1.rlib(tcod-c4ab14fb7f48b4a1.0.o) : error LNK2019: unresolved external symbol __imp_TCOD_random_get_int_mean referenced in function _ZN4tcod6random3Rng12get_int_mean17hf808abccd597c406E
libtcod-c4ab14fb7f48b4a1.rlib(tcod-c4ab14fb7f48b4a1.0.o) : error LNK2019: unresolved external symbol __imp_TCOD_random_get_float referenced in function _ZN4tcod6random3Rng9get_float17hdb772995c5521d7dE
libtcod-c4ab14fb7f48b4a1.rlib(tcod-c4ab14fb7f48b4a1.0.o) : error LNK2019: unresolved external symbol __imp_TCOD_random_get_float_mean referenced in function _ZN4tcod6random3Rng14get_float_mean17hcb6380fbc5207001E
libtcod-c4ab14fb7f48b4a1.rlib(tcod-c4ab14fb7f48b4a1.0.o) : error LNK2019: unresolved external symbol __imp_TCOD_random_get_double referenced in function _ZN4tcod6random3Rng10get_double17h939b3e903a152c5eE
libtcod-c4ab14fb7f48b4a1.rlib(tcod-c4ab14fb7f48b4a1.0.o) : error LNK2019: unresolved external symbol __imp_TCOD_random_get_double_mean referenced in function _ZN4tcod6random3Rng15get_double_mean17hfb809034af856162E
libtcod-c4ab14fb7f48b4a1.rlib(tcod-c4ab14fb7f48b4a1.0.o) : error LNK2019: unresolved external symbol __imp_TCOD_sys_set_fps referenced in function _ZN4tcod6system7set_fps17h949fe75e7128f651E
libtcod-c4ab14fb7f48b4a1.rlib(tcod-c4ab14fb7f48b4a1.0.o) : error LNK2019: unresolved external symbol __imp_TCOD_sys_get_fps referenced in function _ZN4tcod6system7get_fps17ha400ec9f52b94841E
libtcod-c4ab14fb7f48b4a1.rlib(tcod-c4ab14fb7f48b4a1.0.o) : error LNK2019: unresolved external symbol __imp_TCOD_sys_get_last_frame_length referenced in function _ZN4tcod6system21get_last_frame_length17hdcc1e7bfa3f89840E
libtcod-c4ab14fb7f48b4a1.rlib(tcod-c4ab14fb7f48b4a1.0.o) : error LNK2019: unresolved external symbol __imp_TCOD_sys_sleep_milli referenced in function _ZN4tcod6system5sleep17h3ca0dc16258478a9E
libtcod-c4ab14fb7f48b4a1.rlib(tcod-c4ab14fb7f48b4a1.0.o) : error LNK2019: unresolved external symbol __imp_TCOD_sys_elapsed_milli referenced in function _ZN4tcod6system16get_elapsed_time17ha3244302d14e7215E
libtcod-c4ab14fb7f48b4a1.rlib(tcod-c4ab14fb7f48b4a1.0.o) : error LNK2019: unresolved external symbol __imp_TCOD_sys_save_screenshot referenced in function _ZN4tcod6system20save_screenshot_auto17h3b6fc52e88d6c33cE
libtcod-c4ab14fb7f48b4a1.rlib(tcod-c4ab14fb7f48b4a1.0.o) : error LNK2019: unresolved external symbol __imp_TCOD_sys_force_fullscreen_resolution referenced in function _ZN4tcod6system27force_fullscreen_resolution17hb336bd330f966c68E
libtcod-c4ab14fb7f48b4a1.rlib(tcod-c4ab14fb7f48b4a1.0.o) : error LNK2019: unresolved external symbol __imp_TCOD_sys_get_current_resolution referenced in function _ZN4tcod6system22get_current_resolution17h241e2f6754fbedccE
libtcod-c4ab14fb7f48b4a1.rlib(tcod-c4ab14fb7f48b4a1.0.o) : error LNK2019: unresolved external symbol __imp_TCOD_sys_get_fullscreen_offsets referenced in function _ZN4tcod6system21get_fullscreen_offset17he7589e7225ad0576E
libtcod-c4ab14fb7f48b4a1.rlib(tcod-c4ab14fb7f48b4a1.0.o) : error LNK2019: unresolved external symbol __imp_TCOD_sys_get_char_size referenced in function _ZN4tcod6system13get_char_size17h8f9580edd6f8bf34E
libtcod-c4ab14fb7f48b4a1.rlib(tcod-c4ab14fb7f48b4a1.0.o) : error LNK2019: unresolved external symbol __imp_TCOD_sys_clipboard_get referenced in function _ZN4tcod6system13get_clipboard17h37073a9ea548622dE
C:\dev\games\rusty-roguelike\target\release\deps\rusty_roguelike-bcf554dbd8bb6d21.exe : fatal error LNK1120: 114 unresolved externals

error: aborting due to previous error

error: Could not compile `rusty-roguelike`.

Caused by:
  process didn't exit successfully: `rustc --crate-name rusty_roguelike src\main.rs --crate-type bin -C opt-level=3 -C metadata=bcf554dbd8bb6d21 -C extra-filename=-bcf554dbd8bb6d21 --out-dir C:\dev\games\rusty-roguelike\target\release\deps --emit=dep-info,link -L dependency=C:\dev\games\rusty-roguelike\target\release\deps --extern rand=C:\dev\games\rusty-roguelike\target\release\deps\librand-f3e5211f56db4055.rlib --extern tcod=C:\dev\games\rusty-roguelike\target\release\deps\libtcod-c4ab14fb7f48b4a1.rlib -L native=X:\Programs\Visual Studio 15\VC\atlmfc\lib\amd64 -L native=C:\dev\games\rusty-roguelike\target\release\build\tcod-sys-ee576a2106520bdb\out -L C:\dev\games\rusty-roguelike\target\release\build\tcod-sys-ee576a2106520bdb\out -L C:\dev\games\rusty-roguelike\target\release\build\tcod-sys-ee576a2106520bdb\out` (exit code: 101)
```

--- a/tcod_sys/build.rs
+++ b/tcod_sys/build.rs
@@ -3,7 +3,7 @@ extern crate pkg_config;
 
 use std::env;
 use std::fs;
-use std::path::Path;
+use std::path::{Path, PathBuf};
 
 
 fn build_libtcod_objects(mut config: gcc::Config, sources: &[&str]) {
@@ -68,7 +68,6 @@ fn main() {
 	    "libtcod/src/zlib/gzread.c",
 	    "libtcod/src/zlib/gzwrite.c",
     ];
-    gcc::compile_library("libz.a", libz_sources);
 
     let libtcod_sources = &[
  	    "libtcod/src/bresenham_c.c",
@@ -103,6 +102,8 @@ fn main() {
     ];
 
     if target.contains("linux") {
+        gcc::compile_library("libz.a", libz_sources);
+
         // Build the *.o files:
         {
             let mut config = gcc::Config::new();
@@ -136,6 +137,8 @@ fn main() {
         pkg_config::find_library("x11").unwrap();
 
     } else if target.contains("darwin") {
+        gcc::compile_library("libz.a", libz_sources);
+
         // Build the *.o files
         {
             let mut config = gcc::Config::new();
@@ -175,6 +178,8 @@ fn main() {
         assert!(sdl_lib_dir.is_dir());
         assert!(sdl_include_dir.is_dir());
         fs::copy(&sdl_lib_dir.join("SDL.dll"), &dst.join("SDL.dll")).unwrap();
+
+        gcc::compile_library("libz.a", libz_sources);
 
         // Build the *.o files:
         {
@@ -217,36 +222,36 @@ fn main() {
         assert!(sdl_lib_dir.is_dir());
         assert!(sdl_include_dir.is_dir());
         fs::copy(&sdl_lib_dir.join("SDL.dll"), &dst.join("SDL.dll")).unwrap();
-        fs::copy(&sdl_lib_dir.join("SDL.lib"), &dst.join("SDL.dll.lib")).unwrap();
-
-        // Build the *.o files
-        {
-            let mut config = gcc::Config::new();
-            config.include(sdl_include_dir.to_str().unwrap());
-            config.define("LIBTCOD_EXPORTS", None);
-            config.define("NO_OPENGL", None);
-            build_libtcod_objects(config, libtcod_sources);
-        }
+        fs::copy(&sdl_lib_dir.join("SDL.lib"), &dst.join("SDL.lib")).unwrap();
+        fs::copy(&sdl_lib_dir.join("SDLmain.lib"), &dst.join("SDLmain.lib")).unwrap();
 
         // Build the DLL
         let mut config = gcc::Config::new();
-        for c_file in libtcod_sources {
-            config.flag(dst.join(c_file).with_extension("o").to_str().unwrap());
+        config.flag("/DLIBTCOD_EXPORTS");
+        config.flag("/DNO_OPENGL");
+        config.include(sdl_include_dir.to_str().unwrap());
+        config.include(Path::new("libtcod").join("src").join("zlib"));
+        config.include(Path::new("libtcod").join("include"));
+        for c_file in libz_sources.iter().chain(libtcod_sources) {
+            // Make sure the path is in the Windows format. This
+            // shouldn't matter but it's distracting when debugging
+            // build script issues.
+            let path = c_file.split('/').fold(PathBuf::new(), |path, segment| path.join(segment));
+            config.flag(src.join(path).to_str().unwrap());
         }
-        config.flag(dst.join("libz.a").to_str().unwrap());
         config.flag("User32.lib");
-        config.flag("SDL.dll.lib");
-        config.flag("/LD");
+        config.flag("SDL.lib");
+        config.flag("SDLmain.lib");
         config.flag("/link");
         config.flag(&format!("/LIBPATH:{}", dst.to_str().unwrap()));
         config.flag("/DLL");
-        config.flag(&format!("/OUT:{}", dst.join("libtcod.dll").display()));
+        config.flag(&format!("/OUT:{}", dst.join("tcod.dll").display()));
 
         compile_config(config);
-        assert!(dst.join("libtcod.dll").is_file());
+        assert!(dst.join("tcod.dll").is_file());
 
         println!("cargo:rustc-link-search={}", dst.display());
-        println!("cargo:rustc-link-lib=dylib=SDL.dll");
+        println!("cargo:rustc-link-lib=dylib=SDL");
         println!("cargo:rustc-link-lib=User32");
     }
 

